### PR TITLE
Call get metadata after error to discover new topic created by another consumer or producer

### PR DESCRIPTION
**Pull request by KillianG**
Issue: BB-453

### CODE REVIEW

File: lib/BackbeatConsumer.js [hunk 1]
```diff
@@ -424,6 +426,18 @@ class BackbeatConsumer extends EventEmitter {
...
+                if (err && this._consumer.isConnected() && !this.isPaused()) {
+                    this._consumer.getMetadata({ topic: this._topic }, err => {
+                        if (err) {
+                            this._log.error('error getting kafka metadatas', {
+                                error: err,
+                                topic: this._topic,
+                                groupId: this._groupId,
+                            });
+                            this.emit('error', err);
+                        }
```
Comment (francoisferrand) [inline lib/BackbeatConsumer.js]: do we really care ?
if there is another error, we should already catch it ; and if there is no error, we don't really care that `getMetadata` itself failed... (this is kind of a probabilistic workaround)

File: lib/BackbeatConsumer.js [hunk 2]
```diff
@@ -180,6 +180,8 @@ class BackbeatConsumer extends EventEmitter {
             // automatically create topic
             'allow.auto.create.topics': true,
             'rebalance_cb': true,
+            // decrease max metadata age to 5s to avoid lag
```
Comment (francoisferrand) [inline lib/BackbeatConsumer.js]: "lag" does not really help in this context : you need to explain what _lag_ it avoids...

**Comment by francoisferrand** [comment]
this should go on 8.6, this is bugfix

**Review by jonathan-gramain — APPROVED** [review]
LGTM, but couldn't it be better as a solution to this issue to pre-create the archive topic with the wanted number of partitions, e.g. at deployment time, or when the process that produces to this topic starts?

**Comment by KillianG** [comment]
> LGTM, but couldn't it be better as a solution to this issue to pre-create the archive topic with the wanted number of partitions, e.g. at deployment time, or when the process that produces to this topic starts?

I don't think so, because in the case the customer wants to add more partitions to a topic, we don't want the default 10mn lag..

**Review by benzekrimaha — APPROVED** [review]

**Comment by francoisferrand** [comment]
> couldn't it be better as a solution to this issue to pre-create the archive topic with the wanted number of partitions, e.g. at deployment time, or when the process that produces to this topic starts?

yes, this is (kind of) planned. We already started managing topics from ZKOP, and will finish the effort in next quarter(s): e.g. covering the few topics (esp. cold) which are not yet managed there, then disabling the creating of the topics by consumers & producers.

**Pull request merged**